### PR TITLE
(doc) Update docs for v2.3.0

### PR DESCRIPTION
**Pull request by gep13**
## Description Of Changes

The changes here are as a result of running GenerateDocs.ps1, and also some manual changes to the source files in CLI and CLE to try to bring things into line.

## Motivation and Context

With the upcoming release of Chocolatey CLI v2.3.0, there are some changes to the docs needed.

## Testing

* [x] I have previewed these changes using the [Docker Container](https://github.com/chocolatey/docs/tree/master/.devcontainer) or another method before submitting this pull request.

## Change Types Made

* [x] Minor documentation fix (typos etc.).
* [ ] Major documentation change (refactoring, reformatting or adding documentation to existing page).
* [ ] New documentation page added.
* [ ] The change I have made should have a video added, and I have raised an issue for this.
    * Issue #

## Change Checklist

* [ ] Requires a change to menu structure (top or left-hand side)/
* [ ] Menu structure has been updated

## Related Issue

N/A

### CODE REVIEW

File: input/en-us/choco/commands/rule.md
Comment (corbob) [inline input/en-us/choco/commands/rule.md:1]: Has the `ch0.co` shorturl been generated for this new command? Do we need to generate it?

File: input/en-us/create/functions/index.md
Comment (corbob) [inline input/en-us/create/functions/index.md:1]: It looks like the encoding on this file changed. Also: Are we intentionally removing reference to the functions that have been updated to be cmdlets?

File: input/en-us/choco/commands/rule.md
Comment (gep13) [reply input/en-us/choco/commands/rule.md:1]: Yes, I have created it.

File: input/en-us/create/functions/index.md
Comment (gep13) [reply input/en-us/create/functions/index.md:1]: I will get the encodings fixed up.

Yes, @vexx32 has another PR here: https://github.com/chocolatey/docs/pull/993 for making a split between Functions and Cmdlets.

File: input/en-us/choco/commands/feature.md [hunk 1]
```diff
@@ -40,6 +40,11 @@ Exit codes that normally result from running this command.
 Normal:
  - 0: operation was successful, no issues detected
  - -1 or 1: an error has occurred
+ - 2: nothing to do, feature already set (enhanced)
+
+> :choco-info: **NOTE** Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes'
+ turned on, then choco will provide enhanced exit codes that allow
+ better integration and scripting.
```
Comment (st3phhays) [inline input/en-us/choco/commands/feature.md:47]: These new alerts will not be rendered correctly on the website.  I believe I have this working in my Astro changes I have made, but I'll double check that.

```suggestion
> :choco-info: **NOTE**
> 
> Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes' turned on, then choco will provide enhanced exit codes that allow better integration and scripting.

```

File: input/en-us/choco/commands/pin.md [hunk 2]
```diff
@@ -44,6 +44,11 @@ Exit codes that normally result from running this command.
 Normal:
  - 0: operation was successful, no issues detected
  - -1 or 1: an error has occurred
+ - 2: nothing to do (enhanced)
+
+> :choco-info: **NOTE** Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes'
+ turned on, then choco will provide enhanced exit codes that allow
+ better integration and scripting.
```
Comment (st3phhays) [inline input/en-us/choco/commands/pin.md:51]:
```suggestion
> :choco-info: **NOTE**
> 
> Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes' turned on, then choco will provide enhanced exit codes that allow better integration and scripting.

```

File: input/en-us/choco/commands/setapikey.md [hunk 3]
```diff
@@ -70,6 +70,11 @@ Exit codes that normally result from running this command.
 Normal:
  - 0: operation was successful, no issues detected
  - -1 or 1: an error has occurred
+ - 2: nothing to do, apikey already set (enhanced)
+
+> :choco-info: **NOTE** Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes'
+ turned on, then choco will provide enhanced exit codes that allow
+ better integration and scripting.
```
Comment (st3phhays) [inline input/en-us/choco/commands/setapikey.md:77]:
```suggestion
> :choco-info: **NOTE**
> 
> Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes' turned on, then choco will provide enhanced exit codes that allow better integration and scripting.

```

File: input/en-us/choco/commands/sources.md [hunk 4]
```diff
@@ -48,6 +48,11 @@ Exit codes that normally result from running this command.
 Normal:
  - 0: operation was successful, no issues detected
  - -1 or 1: an error has occurred
+ - 2: nothing to do (enhanced)
+
+> :choco-info: **NOTE** Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes'
+ turned on, then choco will provide enhanced exit codes that allow
+ better integration and scripting.
```
Comment (st3phhays) [inline input/en-us/choco/commands/sources.md:55]:
```suggestion
> :choco-info: **NOTE**
> Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes' turned on, then choco will provide enhanced exit codes that allow better integration and scripting.

```

File: input/en-us/choco/commands/unpackself.md [hunk 5]
```diff
@@ -10,7 +10,9 @@ RedirectFrom:
 
 <!-- This file is automatically generated based on output from https://github.com/chocolatey/choco/blob/master/src/chocolatey/infrastructure.app/commands/ChocolateyUnpackSelfCommand.cs using https://github.com/chocolatey/choco/blob/master/GenerateDocs.ps1. Contributions are welcome at the original location(s). If the file is not found, it is not part of the open source edition of Chocolatey or the name of the file is different. --> 
 
-# UnpackSelf Command (choco unpackself)
+# [DEPRECATED] UnpackSelf Command (choco unpackself)
+
+> :choco-info: **NOTE** Unpackself has been deprecated and will be removed in version 3.0.0.
```
Comment (st3phhays) [inline input/en-us/choco/commands/unpackself.md:15]:
```suggestion
> :choco-info: **NOTE**
>
> Unpackself has been deprecated and will be removed in version 3.0.0. 

```

File: input/en-us/choco/commands/upgrade.md [hunk 6]
```diff
@@ -62,6 +62,11 @@ Exit codes that normally result from running this command.
 Normal:
  - 0: operation was successful, no issues detected
  - -1 or 1: an error has occurred
+ - 2: nothing to do, no packages outdated (enhanced)
+
+> :choco-info: **NOTE** Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes'
```
Comment (st3phhays) [inline input/en-us/choco/commands/upgrade.md:67]:
```suggestion
> :choco-info: **NOTE**
> 
> Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes'

```

File: input/en-us/create/commands/api-key.md [hunk 7]
```diff
@@ -70,6 +70,11 @@ Exit codes that normally result from running this command.
 Normal:
  - 0: operation was successful, no issues detected
  - -1 or 1: an error has occurred
+ - 2: nothing to do, apikey already set (enhanced)
+
+> :choco-info: **NOTE** Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes'
+ turned on, then choco will provide enhanced exit codes that allow
+ better integration and scripting.
```
Comment (st3phhays) [inline input/en-us/create/commands/api-key.md:77]:
```suggestion
> :choco-info: **NOTE**
> 
> Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes' turned on, then choco will provide enhanced exit codes that allow better integration and scripting.

```

File: input/en-us/choco/commands/config.md [hunk 8]
```diff
@@ -42,6 +42,11 @@ Exit codes that normally result from running this command.
 Normal:
  - 0: operation was successful, no issues detected
  - -1 or 1: an error has occurred
+ - 2: nothing to do, config already set/unset (enhanced)
+
+> :choco-info: **NOTE** Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes'
+ turned on, then choco will provide enhanced exit codes that allow
+ better integration and scripting.
```
Comment (st3phhays) [inline input/en-us/choco/commands/config.md:49]:
```suggestion
> :choco-info: **NOTE**
>
> Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes' turned on, then choco will provide enhanced exit codes that allow better integration and scripting.

```

File: input/en-us/choco/commands/features.md [hunk 9]
```diff
@@ -42,6 +42,11 @@ Exit codes that normally result from running this command.
 Normal:
  - 0: operation was successful, no issues detected
  - -1 or 1: an error has occurred
+ - 2: nothing to do, feature already set (enhanced)
+
+> :choco-info: **NOTE** Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes'
+ turned on, then choco will provide enhanced exit codes that allow
+ better integration and scripting.
```
Comment (st3phhays) [inline input/en-us/choco/commands/features.md:49]:
```suggestion
> :choco-info: **NOTE**
> 
> Starting in v2.3.0, if you have the feature 'useEnhancedExitCodes' turned on, then choco will provide enhanced exit codes that allow better integration and scripting.

```

**Review by st3phhays — CHANGES_REQUESTED** [review]
I noticed some of the new notes that were added are not in the right format. I'm unsure why these were not caught by the GenerateDocs.ps1 file, but I *think I have this working in my Astro branch. I will confirm that soon.

File: input/en-us/create/functions/index.md
Comment (gep13) [reply input/en-us/create/functions/index.md:1]: @corbob that change here was to move from UTF-8 to UTF-8 with BOM, so this is a valid change from what I can see.

**Comment by gep13** [comment]
@corbob @st3phhays this is ready for another review.  Still not ready to merge, but it you are happy, leave an approval and I can merge when getting ready for the release.

**Comment by gep13** [comment]
@JPRuskin I know that this is in draft, but would you mind having a look over this, and hopefully give a final approval before we release? Thanks!

**Review by JPRuskin — COMMENTED** [review]
LGTM, some weirdness around eol-whitespace being brought into markdown (might trigger some rules later).

**Comment by gep13** [comment]
@corbob @st3phhays @JPRuskin thank you all for the reviews here!  Going to merge this once the CI build finishes

**Pull request merged**